saleor/webhook/utils.py

```diff
@@ -54,13 +48,90 @@ def get_webhooks_for_event(
     webhook_events = WebhookEvent.objects.using(
         settings.DATABASE_CONNECTION_REPLICA_NAME
     ).filter(event_type__in=event_types)
+    return (
+        Q(is_active=True)
+        & Q(Exists(apps.filter(id=OuterRef("app_id"))))
+        & Q(Exists(webhook_events.filter(webhook_id=OuterRef("id"))))
+    )
+
+
+def get_webhooks_for_event(
+    event_type: str,
+    webhooks: Optional["QuerySet[Webhook]"] = None,
+    apps_ids: Optional["list[int]"] = None,
+    apps_identifier: Optional[list[str]] = None,
+) -> "QuerySet[Webhook]":
+    """Get active webhooks from the database for an event."""
+
+    if webhooks is None:
+        # For this QS replica usage is applied later, as this QS could be also passed
+        # as parameter.
+        webhooks = Webhook.objects.all()
+
+    filters = get_filter_for_single_webhook_event(
+        event_type=event_type, apps_ids=apps_ids, apps_identifier=apps_identifier
+    )
+
     return (
         webhooks.using(settings.DATABASE_CONNECTION_REPLICA_NAME)
-        .filter(
-            Q(is_active=True)
-            & Q(Exists(apps.filter(id=OuterRef("app_id"))))
-            & Q(Exists(webhook_events.filter(webhook_id=OuterRef("id"))))
+        .filter(filters)
+        .select_related("app")
+        .prefetch_related("app__permissions__content_type")
+    )
+
+
+def get_webhooks_for_multiple_events(
+    event_types: list[str],
+    webhooks: Optional["QuerySet[Webhook]"] = None,
+    apps_ids: Optional["list[int]"] = None,
+    apps_identifier: Optional[list[str]] = None,
+) -> dict[str, list[Webhook]]:
+    if webhooks is None:
+        # For this QS replica usage is applied later, as this QS could be also passed
+        # as parameter.
+        webhooks = Webhook.objects.all()
+
+    filter_expresion = Q()
+    for event_type in set(event_types):
+        filter_expresion |= Q(
```

Comment:
If there were many events, this query could result in a quite complex query plan, I'm afraid. Maybe it won't be the case for now, as I guess we won't have many webhooks or events at once. Anyway, I think in the long term, a dedicated SQL query could be better to handle cases with multiple events than joining filters from the single-event case.

Reply:
You're right, if we want to change this, I would need to fetch all apps, all their permissions, then all webhooks assigned to these apps, and all webhook events assigned to webhook. Then I could get the same result as |= filter_expression. 
I am working on the order part right now, so I will try to figure out if I can improve this right now.

Reply:
Added this to my to-do list for the task: https://linear.app/saleor/issue/MERX-156/application-with-order-updated-webhook-subscribe-for-ordertotal-may
I will double check if we can make this call better/cost of current query in case of having multiple webhooks